projects/odf_transform/odf_transform/odf_to_netcdf.py

```diff
@@ -49,72 +49,64 @@ def write_ctd_ncfile(
     ds.attrs.update(odf.global_attributes_from_header(metadata))  # From ODF header
 
     # Add New Variables
-    ds = odf.generate_variables_from_header(ds, metadata, config['cdm_data_type'])  # From ODF header
+    metadata["variable_attributes"] = odf.odf_flag_variables(
+        metadata["variable_attributes"], config.get("flag_convention")
+    )
+    ds = odf.generate_variables_from_header(
+        ds, metadata, config["cdm_data_type"]
+    )  # From ODF header
     # geographic_area
-    ds["geographic_area"] = get_geo_code([ds["longitude"].mean(), ds["latitude"].mean()], polygons)
+    ds["geographic_area"] = get_geo_code(
+        [ds["longitude"].mean(), ds["latitude"].mean()], polygons
+    )
 
     # Add file name as a variable
-    if config.get('cdm_data_type') == 'Profile':
-        ds['profile_id'] = os.path.split(odf_path)[-1]
-    elif config.get('cdm_data_type') == 'TimeSeries':
-        ds['timeseries_id'] = os.path.split(odf_path)[-1]
-    elif config.get('cdm_data_type') == 'Trajectory':
-        ds['trajectory_id'] = os.path.split(odf_path)[-1]
+    ds["file_id"] = os.path.split(odf_path)[-1]
+    if config.get("cdm_data_type") == "Profile":
+        ds["profile_id"] = os.path.split(odf_path)[-1]
```

Comment:
instead of duplicating `os.path.split(odf_path)[-1]` and `config.get("cdm_data_type")`, could be more clear if you use variables, eg `cdm_data_type=config.get("cdm_data_type")` and `mission_id=os.path.split(odf_path)[-1]`

Reply:
Sure I'll do that!